test/onnx/verify/dynamicquantizelinear_1d_test.cpp

```diff
@@ -79,3 +80,61 @@ TEST_CASE(dynamicquantizelinear_1d_max_adjusted_test)
     std::vector<uint8_t> y_zpt_gold = {255};
     EXPECT(migraphx::verify::verify_rms_range(y_zpt, y_zpt_gold));
 }
+
+TEST_CASE(dynamicquantizelinear_1d_dot_simplify_test)
+{
+    auto p   = migraphx::parse_onnx("dynamicquantizelinear_1d_dot_test.onnx");
+    auto* mm = p.get_main_module();
+    migraphx::run_passes(*mm, {migraphx::simplify_dynamicquantizelinear{}});
```

Comment:
`dot` if it doesn't allow different types for the inputs then you should also put this pass inside `ref/target.cpp` directly.

Reply:
If I add this to ref target, it will break the other tests that expect the output of dynamicquantizelinear to be uint8 not int8.

Reply:
Ive added back the same_types() check as part of dot. its required me to splice the change as part of the pass instead